app/src/main/java/com/guruthedev/instagram/MainActivity.kt

```diff
@@ -24,12 +32,46 @@ class MainActivity : AppCompatActivity() {
     val CHANNEL_NAME = "channelName"
     val NOTIF_ID = 0
 
+
     @RequiresApi(Build.VERSION_CODES.M)
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
         binding = DataBindingUtil.setContentView(this, R.layout.activity_main)
         createNotificationChannel()
         updatedPendingIntent()
+
+        val cm: ConnectivityManager =
```

Comment:
the code for connectivityManager can be moved to a separate file, make it a reusable code.
**NOTE: UI should not hold any extra code apart from displaying or updating the UI.**